assume/units/dsm_load_shift.py

```diff
@@ -6,6 +6,44 @@
 
 
 class DSMFlex:
+    def initialize_components(self, components: dict[str, dict]):
+        """
+        Initializes the DSM components by creating and adding blocks to the model.
+
+        This method iterates over the provided components, instantiates their corresponding classes,
+        and adds the respective blocks to the Pyomo model.
+
+        Args
+        ----
+            components (dict[str, dict]): A dictionary where each key is a technology name and
+                                        the value is a dictionary of parameters for the respective technology.
+                                        Each technology is mapped to a corresponding class in `demand_side_components`.
+
+        The method:
+        ----------
+        - Looks up the corresponding class for each technology in `demand_side_components`.
+        - Instantiates the class by passing the required parameters.
+        - Adds the resulting block to the model under the `dsm_blocks` attribute.
+        """
+        from assume.units import demand_side_components
```

Comment:
This shadows a circular import, we should move the dictionary of available demand_side_components to something like `assume.units.optimization` ?